content/developers/developer-patterns/navigating-merklelogs/index.md

```diff
@@ -0,0 +1,660 @@
+This term is due to the name of the verifiable data structure used for the log: An MMR or "Merkle Mountain Range" [^1].
+
+[^1]: Merkle Mountain Ranges have seen extensive use in systems that need long term tamper evident storage, notably [zcash](https://zips.z.cash/zip-0221), [mimblewimble](), and [many others](https://zips.z.cash/zip-0221#additional-reading).
+Merkle Mountain Ranges are attributed to [Peter Todd](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-May/012715.html), though much parallel invention has occurred. The "post order", write once & append only nature of the tree organisation was first discussed in 3.3 of the [Crosby, Wallach paper](https://static.usenix.org/event/sec09/tech/full_papers/crosby.pdf).
+They have been independently analyzed in the context of [cryptographic asynchronous accumulators](https://eprint.iacr.org/2015/718.pdf), Generalized multi-proofs for [Binary Numeral Trees](https://eprint.iacr.org/2021/038.pdf).
+And also by the [ethereum research community](https://ethresear.ch/t/batching-and-cyclic-partitioning-of-logs/536).
+
+Each massif contains the verifiable data for a fixed number, and sequential range, of your events.
+The number of events is determined by your log configuration parameter `massif height`.
+
+Currently, all logs have a massif height of `14` And the number of event *leaf* log entries in each massif is 2<sup>height-1</sup>, which is 2<sup>14-1</sup> leaves, which is `8192` leaves. 14 was picked because it affords a massif storage size of comfortably under 4mb (less than 2 actually), and for many situations it is practical to handle this size as "one lump". We discuss [reconfiguration here](#the-massif-height-is-constant-for-all-blobs-in-a-log-configuration)
+
+
+Above, we have drawn `massif n` as open ended to illustrate that the last massif is always in the process of being *appended* to.
+
+[Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) gives you a shortcut for picking the right massif.
+It can also be fairly easily computed from only the `merklelog_entry.commit.index` *mmrIndex* on your event using the example javascript on that page.
+
+Here we deal with the format of a single massif.
+
+## Every massif blob is a series of 32 byte aligned fields
+
+Every massif in a log is structured as a series of `32` byte fields.
+All individual entries in the log are either exactly 32 bytes or a small multiple of `32`.
+
+```
+  0               32
+  +----------------+
+  |                | field 0
+  +----------------+
+  .                .
+  .                .
+  +----------------+
+  |                | field n
+  +----------------+
+```
+
+Given a tenant identity of `72dc8b10-dde5-43fe-99bc-16a37fd98c6a` that tenants first massif blob can be found at:
+
+**https://app.datatrails.ai/verifiabledata/merklelogs/v1/mmrs/tenant/72dc8b10-dde5-43fe-99bc-16a37fd98c6a/0/massifs/0000000000000000.log**
```

Comment:
I don't think we have content types sorted such that it can be clickable. If it can be clickable that is my preference too. If it *cant* be clickable, I would prefer it was a code block `https://app.datatrails.ai/verifiabledata/merklelogs/v1/mmrs/tenant/72dc8b10-dde5-43fe-99bc-16a37fd98c6a/0/massifs/0000000000000000.log`, and possibly a curl rune, so that readers can always confirm directly as they read.